packages/trie/src/baseTrie.ts

```diff
@@ -561,16 +569,29 @@ export class Trie {
   ): Buffer | (EmbeddedNode | null)[] {
     const rlpNode = node.serialize()
 
+    /**
+     * TODO: analyze why this length check is here, 2021-04-29
+     *
+     * Hint from https://eth.wiki/fundamentals/patricia-tree:
+     * "When one node is referenced inside another node, what is included is H(rlp.encode(x)),
+     * where H(x) = sha3(x) if len(x) >= 32 else x and rlp.encode is the RLP encoding function."
+     *
+     * If removed various tests fail. However this prevents certain delete operations
+     * when deleteFromDB is activated.
+     * Related test: `index.spec.ts`, "setting back state root (deleteFromDB)"
+     */
```

Comment:
Hey!
So, as mentioned in the docs you linked, I believe the length check is necessary because the `key` is only the `hashRoot` if the RLP encoded `value` is >= 32. 

And actually, in the case where `rlpNode.length < 32`, I don't think we even store a key-value pair, since the value is just referenced directly (`H(x)` is reversible if `rlpNode.length < 32`) so it's not necessary to store anything, which makes me think it would just not get added to the trie in the first place.